org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/breakpoints/ConditionalBreakpointHandler.java

```diff
@@ -217,12 +220,15 @@ public int breakpointHit(IJavaThread thread, IJavaBreakpoint breakpoint) {
 						.getTopStackFrame();
 				IJavaProject project = lineBreakpoint.getJavaProject(frame);
 				if (project == null) {
-					fireConditionHasErrors(
-							lineBreakpoint,
-							new Message[] { new Message(
-									JDIDebugBreakpointMessages.JavaLineBreakpoint_Unable_to_compile_conditional_breakpoint___missing_Java_project_context__1,
-									-1) });
-					return SUSPEND;
+					IMarker marker = breakpoint.getMarker(); // my changes
+					if (marker != null) {
+						IResource res = marker.getResource();
+						project = JavaCore.create(res.getProject());
```

Comment:
the javadoc of org.eclipse.core.resources.IResource.getProject() says it may retun null. Better catch that case?

Reply:
Sure